pkg/rbac/clusterScopedAuthorization.go

```diff
@@ -0,0 +1,73 @@
+package rbac
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/doug-martin/goqu/v9"
+	"github.com/stolostron/search-v2-api/pkg/config"
+	"k8s.io/klog/v2"
+)
+
+type sharedList struct {
+	err       error
+	resources map[string][]string
+	updatedAt time.Time
+}
+
+func (cache *Cache) checkUserResources(uid string) (bool, error) {
+
```

Comment:
We need a lock here to ensure that only 1 process updates the cache.  Any other processes will wait here until the data is updated.